test/test_automl/test_automl.py

```diff
@@ -933,3 +935,184 @@ def predict(self, x):
         _model_predict(model, X, task, logger=None)
 
         assert len(w) == 1, "One warning sould have been emmited"
+
+
+@pytest.mark.parametrize("dataset_compression", [False])
+def test_param_dataset_compression_false(dataset_compression: bool) -> None:
+    """
+    Parameters
+    ----------
+    dataset_compression: bool
+        The dataset_compression arg set as False
+
+    Expects
+    -------
+    * Should set the private attribute to None
+    """
+    auto = AutoMLRegressor(
+        time_left_for_this_task=30,
+        per_run_time_limit=5,
+        dataset_compression=dataset_compression
+    )
+
+    assert auto._dataset_compression is None
+
+
+@pytest.mark.parametrize("dataset_compression", [True])
```

Comment:
Why is this a new test and not in the one above?

Reply:
It could be technically, mapping `False -> None` and `True -> True`. It does introduce an if statement into the test which I'm a fan of trying to avoid now.

An if statement that branches to two different asserts generally means you are testing two different but related things.